### PR TITLE
Allow tests to be run from Eclipse (#2253)

### PR DESCRIPTION
**Pull request by jskov**
Adds additional test/application lookup paths to support Eclipse's build
output locations.

### CODE REVIEW

File: test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java [hunk 1]
```diff
@@ -21,13 +21,28 @@
...
 public final class PathTestHelper {
-
-    private static final String TEST_CLASSES_FRAGMENT_MAVEN = File.separator + "test-classes";
-    private static final String CLASSES_FRAGMENT_MAVEN = File.separator + "classes";
-    private static final String TEST_CLASSES_FRAGMENT_GRADLE = "classes" + File.separator + "java" + File.separator + "test";
-    private static final String CLASSES_FRAGMENT_GRADLE = "classes" + File.separator + "java" + File.separator + "main";
+    private static final Map<String, String> TEST2MAIN_DIR_FRAGMENTS = new HashMap<>();
+    static {
+        // eclipse
+        TEST2MAIN_DIR_FRAGMENTS.put( //eclipse
```
Comment (gsmet) [inline test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java]: You don't need the comment at the end of the line: you have one just above.

Comment (gsmet) [inline test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java]: I would go with `TEST_TO_MAIN...`.

Comment (gsmet) [inline test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java]: Do we need `bin` for Eclipse? AFAIK, it's following the Maven/Gradle conventions when importing a Maven/Gradle project?

**Review by gsmet — COMMENTED** [review]
I added a couple of questions and comments. I'd like to understand why this is necessary.

(That being said, I'm leaning towards merging it even without the Eclipse part)

File: test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java (hunk 1 above)
Comment (jskov) [reply test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java]: Hi, sorry for the delay in answering - life and work interfered

In my experience, it always uses bin/main, but I use a custom Gradle plugin at work (and run eclipse task manually at home).

So I just tried importing the quarkus-quickstarts/getting-started project with Eclipse's builtin Gradle support. And it also uses bin/main as output.

Makes sense, or Eclipse and Gradle would step on each other's toes.

I will adjust for the two other comments and resubmit.

Cheers!

**Pull request closed without merging**